stdlib/fun.mli

```diff
@@ -20,19 +20,42 @@
 (** {1:combinators Combinators} *)
 
 external id : 'a -> 'a = "%identity"
-(** [id] is the identity function. For any argument [x], [id x] is [x]. *)
+(** [id] is the identity function. For any argument [x], [id x] is [x].
+
+    example:
+    [List.init 3 id] is [[0; 1; 2]] *)
 
 val const : 'a -> (_ -> 'a)
 (** [const c] is a function that always returns the value [c]. For any
-    argument [x], [(const c) x] is [c]. *)
+    argument [x], [(const c) x] is [c].
+
+    example:
+    [List.init 3 (const 0)] is [[0; 0; 0]]
+
+    One important difference between [const x] and [fun _ -> x] in the above
+    example is that [x] is evaluated only once, not three times. *)
 
 val flip : ('a -> 'b -> 'c) -> ('b -> 'a -> 'c)
 (** [flip f] reverses the argument order of the binary function
-    [f]. For any arguments [x] and [y], [(flip f) x y] is [f y x]. *)
+    [f]. For any arguments [x] and [y], [(flip f) x y] is [f y x].
+
+    example:
+    [List.map (flip (-) 2) [4; 6; 8]] is [[2; 4; 6]] *)
```

Comment:
Honestly I find `flip (-) 2` unreadable. Applying non-commutative operators as functions gives code that is very hard to read for people who don't do it on a regular basis. I find `List.sort (Fun.flip compare)` easier to sell.

Reply:
That's not a bad example at all, I like it.